### DISCUSSION TITLE
About concurrent execution of kernel into SM. I had doubts when doing the experiment.

### DISCUSSION THREAD

**Discussion by HCSR-Lab** [accel-sim/accel-sim-framework#168] [Q&A] [answered]
**Experimental environment：**
-gpgpu_concurrent_kernel_sm 1
-gpgpu_max_concurrent_kernel  8
kernel1<<<1,1>>>
kernel2<<<1,1>>>

**question：**
I printed this variable during gdb debugging.
My understanding of this variable is that since it is concurrent, it should be _0x1e00f40,0x1e5654d, 0,0,0,0,0,_ and each value is the address of the kernel.
<img width="1001" alt="275d30448ddb15c7af49ab30dd4416e" src="https://user-images.githubusercontent.com/38456927/218980539-b4bc6697-cbda-4046-ae61-f7ae89e27ef7.png">

But until the end of the first kernel, the print is  0x1e00f40,0, 0,0,0,0,0

**So how should we understand concurrent kernel execution?**

-------------------------------------------------------------------------------------------------------------------------------------------------
Sorry, I didn't understand the code. After many inquiries, this variable m_running_kernels just gets the kernel address in execution, so there is only one address.
I wonder if I add such modification (-gpgpu_concurrent_kernel_sm 1
-gpgpu_max_concurrent_kernel 8), is there any difference from the case without modification? I found no difference in my experiments.

**Comment by JRPan** [comment, marked as answer]
All kernels launched should show up here. Can you show the gpgpu-sim output?

**Reply by HCSR-Lab** [reply]
![image](https://user-images.githubusercontent.com/38456927/219278097-3e41ad3c-af3c-4a60-a502-c71251f2b2c5.png)

[log-0.txt](https://github.com/accel-sim/accel-sim-framework/files/10752548/log-0.txt)


As you can see, I didn't increase the stream, the two kernels are in one stream.

Therefore, do I need to modify the CUDA program to achieve the condition of running multiple kernels?

**Reply by JRPan** [reply]
Yes, only kernels not on the same stream can run concurrently.

**Reply by HCSR-Lab** [reply]
thank you for your help.
